core/src/test/scala/tech/beshu/ror/unit/acl/factory/decoders/rules/auth/GroupsRuleSettingsTests.scala

```diff
@@ -413,6 +451,73 @@ sealed abstract class GroupsRuleSettingsTests[R <: BaseGroupsRule : ClassTag](ru
                   Mapping(group("group2"), UniqueNonEmptyList.of(GroupId("ldap_group4")))
                 )))
 
+                rule1 shouldBe an[AuthKeyRule]
+                rule1.asInstanceOf[AuthKeyRule].settings should be {
+                  BasicAuthenticationRule.Settings(Credentials(User.Id("cartman"), PlainTextSecret("pass")))
+                }
+                rule2 shouldBe an[ExternalAuthorizationRule]
+                rule2.asInstanceOf[ExternalAuthorizationRule].settings.permittedGroupsLogic should be(
+                  GroupsLogic.Or(PermittedGroupIds(UniqueNonEmptyList.of(GroupId("ldap_group3"), GroupId("ldap_group4"))))
+                )
+              }
+            }
+          )
+        }
+        "advanced groups mapping with structured groups is used" in {
+          assertDecodingSuccess(
+            yaml =
+              s"""
+                 |readonlyrest:
+                 |
+                 |  access_control_rules:
+                 |
+                 |  - name: test_block1
+                 |    ${ruleName.name.value}: [group1, group3]
+                 |
+                 |  users:
+                 |  - username: cartman
+                 |    groups:
```

Comment:
when we consider the advanced group mapping:

the old/basic syntax is:
`- <local_group_id> : [ <external_group_id_pattern> ]`

we should support the old/basic syntax, but the new/extended syntax should be:
````
- local_group: 
    id: <local_group_id> // required
    name: <local_group_name> // required
   external_group_ids: [ <external_group_id_pattern> ] 
```

So two comments here:
1. there is no sense to support the extended/new syntax and optional name - if the user needs to use the name (so they want to distinguish between name and id), they should use the new/extended syntax
2. IMO it's better to add the `local_group` key to make it clear